controllers/core/openstackcontrolplane_controller.go

```diff
@@ -170,10 +172,81 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, nil
 	}
 
-	return r.reconcileNormal(ctx, instance, helper)
+	Log.Info("Looking up the current OpenStackVersion")
+	ctrlResult, err, version := openstack.ReconcileVersion(ctx, instance, helper)
+	if err != nil {
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	// wait until the version is initialized so we have images on the version.Status
+	if version.Status.Conditions.IsFalse(corev1beta1.OpenStackVersionInitialized) {
```

Comment:
I think it would be safer to say `!version.Status.Conditions.IsTrue(...)` so that if it is Unknown or nil then the controller also waits. See https://github.com/openstack-k8s-operators/lib-common/blob/fec16b14c33be8622459fa2a652e5553d4912e68/modules/common/condition/funcs.go#L146-L151